File: library/src/main/java/io/github/anderscheow/validator/rules/BaseRule.java
package io.github.anderscheow.validator.rules;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import io.github.anderscheow.validator.util.ErrorMessage;
import io.github.anderscheow.validator.util.Validate;

public abstract class BaseRule extends ErrorMessage implements Validate {

    public BaseRule() {}

    public BaseRule(@StringRes int errorRes) {
        setErrorRes(errorRes);
    }

    public BaseRule(@NonNull String errorMessage) {
        setErrorMessage(errorMessage);
    }
}

File: library/src/main/java/io/github/anderscheow/validator/rules/BaseRule.kt
package io.github.anderscheow.validator.rules

import androidx.annotation.StringRes
import io.github.anderscheow.validator.interfaces.ErrorImpl
import io.github.anderscheow.validator.interfaces.Validate

abstract class BaseRule : ErrorImpl, Validate {

    constructor()

    constructor(@StringRes errorRes: Int) : super(errorRes)

    constructor(errorString: String) : super(errorString)
}
